core-bundle/src/Resources/contao/library/Contao/Database/Statement.php

```diff
@@ -284,18 +284,19 @@ static function ($varParam)
 		{
 			$this->statement = $this->resConnection->executeQuery($this->strQuery, $arrParams);
 		}
-		catch (DriverException $exception)
+		catch (DriverException|\ArgumentCountError $exception)
 		{
-			if ($arrParams !== array(null))
+			if (!$arrParams || \count($arrParams) <= ($intTokenCount = $this->countQueryTokens()))
```

Comment:
How about moving this logic into the `countQueryTokens()` method as well and renaming it to `isTokenCountMismatch()` or something else? Then we could simplify this condition:

```suggestion
			if ($this->isTokenCountMismatch($arrParams))
```

Reply:
Not possible because `$intTokenCount` is reused in the code below.